source/type/input_output.cpp

```diff
@@ -0,0 +1,63 @@
+/* Copyright © 2001-2019, Canal TP and/or its affiliates. All rights reserved.
+
+This file is part of Navitia,
+    the software to build cool stuff with public transport.
+
+Hope you'll enjoy and contribute to this project,
+    powered by Canal TP (www.canaltp.fr).
+Help us simplify mobility and open public transport:
+    a non ending quest to the responsive locomotion way of traveling!
+
+LICENCE: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+#include "type/input_output.h"
+
+#include "type/connection.h"
+#include "type/dataset.h"
+#include "type/indexes.h"
+#include "type/pt_data.h"
+#include "type/route.h"
+#include "type/serialization.h"
+#include "type/stop_area.h"
+#include "georef/georef.h"
+
+#include <boost/serialization/weak_ptr.hpp>
+
+namespace navitia {
+namespace type {
+template <class Archive>
+void InputOutput::serialize(Archive& ar, const unsigned int /*unused*/) {
+    ar& uri& name& stop_area& coord& idx;
+}
+SERIALIZABLE(InputOutput)
+
+bool InputOutput::operator<(const InputOutput& other) const {
```

Comment:
Out of curiosity : why do you need a comparator ? Do you sort these objects at some point ?

Reply:
All objet (Line, StopArea, ...) into the repository are with this operator. For a sort purpose